app/controllers/posts_controller.rb

```diff
@@ -1,12 +1,13 @@
 # frozen_string_literal: true
 
 class PostsController < ApplicationController
-  before_action :find_post, except: %i[index new create]
+  load_and_authorize_resource
+
+  before_action :set_post, except: %i[index new create]
   before_action :authenticate_user!, except: %i[index show]
-  before_action :owner?, only: %i[edit update destroy]
 
   def index
-    @posts = Post.all.with_rich_text_body_and_embeds
+    @posts = Post.is_public.with_rich_text_body_and_embeds
     @posts = @posts.from_user(params[:user_id]) if params[:user_id]
```

Comment:
Je me demande, si on ne peut pas faire quelque chose de ce genre là ([la documentation](https://github.com/CanCanCommunity/cancancan/blob/develop/docs/check_abilities_mistakes.md)):
```suggestion
    @posts = Poste.accessible_by(current_ability).with_rich_text_body_and_embeds
```